pkg/api/validate.go

```diff
@@ -334,7 +344,11 @@ func validateAgentPoolProfiles(apps []AgentPoolProfile, vnet *net.IPNet) (errs [
 	return
 }
 
-func validateAgentPoolProfile(app AgentPoolProfile, vnet *net.IPNet) (errs []error) {
+func (v *Validator) validateAgentPoolProfile(app AgentPoolProfile, vnet *net.IPNet) (errs []error) {
+	if !v.PluginConfig.TestConfig.RunningUnderTest && isTestVMSize(app.VMSize) {
+		errs = append(errs, fmt.Errorf("invalid VM size %s", app.VMSize))
+	}
+
```

Comment:
wouldn't it be better to turn validMasterAndInfraVMSizes (and friends) into functions that takes the config and returns the correct data? @jim-minter (would we ever have different sizes in different regions or dependent on other options?)

Reply:
agreed.